kyaml/setters2/set_test.go

```diff
@@ -0,0 +1,437 @@
+		input    string
+		expected string
+	}{
+		{
+			name:   "set-replicas",
+			setter: "replicas",
+			openapi: `
+openAPI:
+  definitions:
+    io.k8s.cli.setters.no-match-1':
+      x-k8s-cli:
+        setter:
+          name: no-match-1
+          value: "1"
+    io.k8s.cli.setters.replicas:
+      x-k8s-cli:
+        setter:
+          name: replicas
+          value: "4"
+    io.k8s.cli.setters.no-match-2':
+      x-k8s-cli:
+        setter:
+          name: no-match-2
+          value: "2"
+ `,
+			input: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+spec:
+  replicas: 3 # {"$ref": "#/definitions/io.k8s.cli.setters.replicas"}
+ `,
+			expected: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+spec:
+  replicas: 4 # {"$ref": "#/definitions/io.k8s.cli.setters.replicas"}
+ `,
+		},
+		{
+			name:   "set-arg",
+			setter: "arg1",
+			openapi: `
+openAPI:
+  definitions:
+    io.k8s.cli.setters.replicas:
+      x-k8s-cli:
+        setter:
+          name: replicas
+          value: "4"
+    io.k8s.cli.setters.arg1:
+      x-k8s-cli:
+        setter:
+          name: arg1
+          value: "some value"
+ `,
+			input: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+spec:
+  template:
+    spec:
+      containers:
+      - name: nginx
+        args:
+        - a
+        - b # {"$ref": "#/definitions/io.k8s.cli.setters.arg1"}
+ `,
+			expected: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+spec:
+  template:
+    spec:
+      containers:
+      - name: nginx
+        args:
+        - a
+        - some value # {"$ref": "#/definitions/io.k8s.cli.setters.arg1"}`,
+		},
+		{
+			name:   "substitute-image-tag",
+			setter: "image-tag",
+			openapi: `
+openAPI:
+  definitions:
+    io.k8s.cli.setters.image-name:
+      x-k8s-cli:
+        setter:
+          name: image-name
+          value: "nginx"
+    io.k8s.cli.setters.image-tag:
+      x-k8s-cli:
+        setter:
+          name: image-tag
+          value: "1.8.1"
+    io.k8s.cli.substitutions.image:
+      x-k8s-cli:
+        substitution:
+          name: image
+          pattern: IMAGE_NAME:IMAGE_TAG
+          values:
+          - marker: "IMAGE_NAME"
+            ref: "#/definitions/io.k8s.cli.setters.image-name"
+          - marker: "IMAGE_TAG"
+            ref: "#/definitions/io.k8s.cli.setters.image-tag"
+ `,
+			input: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+spec:
+  template:
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.7.9 # {"$ref": "#/definitions/io.k8s.cli.substitutions.image"}
+ `,
+			expected: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+spec:
+  template:
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.8.1 # {"$ref": "#/definitions/io.k8s.cli.substitutions.image"}
+ `,
+		},
+		{
+			name:   "substitute-annotation",
```

Comment:
How is this case different from the one below "substitute-not-match-setter". My eyes are not able to catch the difference in inputs but outputs are different.